pkg/services/ngalert/api/api_trans_dev.go

```diff
@@ -131,3 +136,140 @@ func (api *API) conditionOldEndpointByID(c *models.ReqContext) response.Response
 
 	return response.JSON(200, evalCond)
 }
+
+// conditionEvalEndpoint handles POST /api/alert-definitions/ruleGroupByOldID.
+func (api *API) ruleGroupByOldID(c *models.ReqContext) response.Response {
+	id := c.ParamsInt64("id")
+	if id == 0 {
+		return response.Error(400, "missing id", nil)
+	}
+
+	save := c.Query("save") == "true"
+
+	getAlert := &models.GetAlertByIdQuery{
+		Id: id,
+	}
+
+	if err := bus.Dispatch(getAlert); err != nil {
+		return response.Error(400, fmt.Sprintf("could find alert with id %v", id), err)
+	}
+
+	if getAlert.Result.OrgId != c.SignedInUser.OrgId {
+		return response.Error(403, "alert does not match organization of user", nil)
+	}
+
+	oldAlert := getAlert.Result
+
+	sb, err := oldAlert.Settings.ToDB()
+	if err != nil {
+		return response.Error(400, "failed to marshal alert settings", err)
+	}
+
+	evalCond, err := translate.DashboardAlertConditions(sb, c.OrgId)
+	if err != nil {
+		return response.Error(400, "Failed to translate alert conditions", err)
+	}
+
+	getDash := &models.GetDashboardQuery{
+		Id:    getAlert.Result.DashboardId,
+		OrgId: getAlert.Result.OrgId,
+	}
+	if err := bus.Dispatch(getDash); err != nil {
+		return response.Error(400, fmt.Sprintf("could find dashboard %v for alert with id %v", getDash.Id, id), err)
+	}
+
+	isGeneralFolder := getDash.Result.FolderId == 0 && !getDash.Result.IsFolder
+
+	var namespaceUID string
+
+	if isGeneralFolder {
+		namespaceUID = "General"
+	} else {
+		getFolder := &models.GetDashboardQuery{
+			Id:    getDash.Result.FolderId,
+			OrgId: getDash.Result.OrgId,
+		}
+		if err := bus.Dispatch(getFolder); err != nil {
+			return response.Error(400, fmt.Sprintf("could find folder %v for alert with id %v", getFolder.Id, id), err)
+		}
+
+		namespaceUID = getFolder.Result.Uid
+	}
+
+	oldNoData := oldAlert.Settings.Get("noDataState").MustString()
+	noDataSetting, err := transNoData(oldNoData)
+	if err != nil {
+		return response.Error(400, "unrecognized no data option", err)
+	}
+
+	oldExecErr := oldAlert.Settings.Get("executionErrorState").MustString()
+	execErrSetting, err := transExecErr(oldExecErr)
+	if err != nil {
+		return response.Error(400, "unrecognized execution error option", err)
+	}
+
+	rule := ngmodels.AlertRule{
+		Title:        oldAlert.Name,
+		Data:         evalCond.Data,
+		Condition:    evalCond.Condition,
+		NoDataState:  noDataSetting,
+		ExecErrState: execErrSetting, // TODO: Translate properly
+
+		// Should Alert rule have interval seconds, or is the group Interval?
+	}
+
+	rgc := apimodels.PostableRuleGroupConfig{
+		// TODO? Generate new name on conflict?
+		Name:     oldAlert.Name,
+		Interval: model.Duration(oldAlert.Frequency * int64(time.Second)),
```

Comment:
This value should be multiple of the [scheduler base interval](https://github.com/grafana/grafana/blob/5c1d4777b2f554ca236c25155ed382764d971a7a/pkg/services/ngalert/ngalert.go#L34) (currently 10 seconds).
Given that the [default](https://grafana.com/docs/grafana/latest/administration/configuration/#min_interval_seconds) minimum interval between rule evaluations for dashboard alerts used to be one second,
we need either to change the scheduler interval or force this value top be greater and exactly divided by scheduler interval.